.github/workflows/markdownlint.yml

```diff
@@ -0,0 +1,32 @@
+name: Markdownlint
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - "**/*.md"
```

Comment:
@yriveiro Not to cause more questions for you... But do you know if it supports [`.mdx` files](https://github.com/openbao/openbao/blob/development/website/content/api-docs/secret/pki.mdx)? All of our API/usage docs are currently using a combination of Markdown and JSX used by Vercel, that could be linted as well. :-)

Reply:
Unfortunately, `markdownlint` doesn't support mdx files https://github.com/DavidAnson/markdownlint/issues/723#issuecomment-1434017183

One way to overcome this could be adding a new step to the workflow that renders mdx files with something like `@mdx-js/mdx`, once we get the output in a temporary folder we run the linting.

Reply:
It seems that what I said in the previous comment is not accurate, indeed `markdownlint` doesn't support mdx files, nevertheless, the action uses `markdownlint-cli2` and this one supports mdx.

The folder `website` will probably throw a lot of warnings as most of the code is generated and that implies that the generator needs to implement the defined rules the community decides first.

Reply:
Ah cool!

From what I recall, I don't think many of the `.mdx` files under `website/` are auto-generated. I know a lot of them were manually maintained, so I don't think we're liable to overwrite them any time soon. That said, I definitely agree, there's a lot of stuff there that is not well formatted and it would be time consuming to fix...

It doesn't look like there's an easy changed-files-only version of this workflow... I think we'd have to look at `GITHUB_BASE_REF` and `GITHUB_SHA` to see which files changed and then look for `**/*.mdx` files (like we do [for the changelog checker](https://github.com/openbao/openbao/blob/development/.github/workflows/changelog-checker.yml)). Perhaps best to ignore them for now then given there are 1k+ of them...

Sorry for the dead-end!